app/views/puppetclasses/_form.html.erb

```diff
@@ -37,6 +37,9 @@
                              :include_blank => "All Environments - (Not filtered)",:class=>'form-control', :onchange=>'filterByEnvironment(this)'%>
             </div>
           </div>
+          <div class='col-md-3'>
```

Comment:
The smart variables tab has: "Help! What is a Smart variable?" as a dismissable alert at the top, while this is a button (see line 67).  I think the button's better for consistency with the rest of the UI, so perhaps replace it on the variables tab?

And yes, I think it might be better right-aligned against the edge of the box rather than the environment selection?  Maybe!